File: app/src/main/java/com/blackbelt/androidboundrv/api/model/PaginatedResponse.java
package com.blackbelt.androidboundrv.api.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

public class PaginatedResponse<T> {

    public static final class Date {

        @Expose
        @Getter
        private String minimum;

        @Expose
        @Getter
        private String maximum;

    }

    @Expose
    @Setter
    @Getter
    private int page;

    @Expose
    @Getter
    @Setter
    @SerializedName("total_pages")
    private int totalPages;

    @Expose
    @Getter
    @Setter
    @SerializedName("total_results")
    private int totalResults;

    @Expose
    @Getter
    List<T> results = new ArrayList<>();

    @Expose
    @Getter
    Date dates;

    public static <T> PaginatedResponse<T> getEmpty() {
        return new PaginatedResponse<>();
    }
}

File: app/src/main/java/com/blackbelt/androidboundrv/api/model/PaginatedResponse.kt
package com.blackbelt.androidboundrv.api.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.ArrayList


data class PaginatedResponse<T>(@Expose var page: Int? = 0,
                                @Expose @SerializedName("total_pages")
                                var totalPages: Int? = 0,
                                @Expose
                                @SerializedName("total_results")
                                var totalResults: Int = 0,
                                @Expose
                                var results: List<T> = ArrayList()) {
    companion object {

        fun <T> getEmpty(): PaginatedResponse<T> {
            return PaginatedResponse()
        }
    }
}